.github/workflows/ubuntu-windows.yml

```diff
@@ -0,0 +1,401 @@
+          asset_content_type: application/zip
+
+
+  build:
+    needs: [build_sirius]
+    name: ${{ matrix.os }} shrd=${{ matrix.shared }} sirius=${{ matrix.sirius }} xprs=${{ matrix.xprs }} extras=${{ matrix.extras }} java=${{ matrix.java }} dotnet=${{ matrix.dotnet }} python=${{ matrix.python }}-${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    env:
+      Sirius_Ref: antares_integration
+      XPRESSDIR: ${{ github.workspace }}/xpressmp813
+      XPRESS: ${{ github.workspace }}/xpressmp813/bin
+      XPRS_LIB_Path: ${{ github.workspace }}/xpressmp813/lib
+      SIRIUS_INSTALL_PATH : ${{ github.workspace }}/sirius_install
+      SIRIUS_BIN_DIR: ${{ github.workspace }}/sirius_install/bin
+      XPRESS_BIN_DIR: ${{ github.workspace }}/xpressmp813/bin
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-2019, ubuntu-20.04]
+        python-version: ['3.7', '3.8', '3.9', '3.10']
+        python: [ON, OFF]
+        xprs: [ON]
+        sirius: [ON, OFF]
+        shared: [ON, OFF]
+        extras: [ON, OFF]
+
+        exclude:
+          - xprs: OFF
+            sirius: OFF
+          - os: windows-2019
+            shared: ON
+          - os: windows-2019
+            shared: OFF
+            extras: OFF
+          - os: ubuntu-20.04
+            shared: OFF
+            extras: ON
+          - os: ubuntu-20.04
+            shared: ON
+            extras: OFF
+          - extras: OFF
+            python: ON
+          - python: OFF
+            python-version: '3.7'
+          - python: OFF
+            python-version: '3.8'
+          - python: OFF
+            python-version: '3.9'
+        include:
+          - extras: ON
+            python: OFF
+            publish-or: ON
+            dotnet: ON
+            java: ON
+          - extras: ON
+            python: ON
+            publish-or: OFF
+            dotnet: OFF
+            java: OFF
+          - extras: OFF
+            python: OFF
+            publish-or: ON
+            java: OFF
+            dotnet: OFF
+
+    steps:
+      - name: set name variables
+        id: names
+        shell: bash
+        run: |
+          SHARED=${{ matrix.shared }}
+          [ $SHARED == "ON" ] && WITH_SHARED="_shared" || WITH_SHARED="_static"
+          XPRS=${{ matrix.xprs }}
+          [ $XPRS == "ON" ] && WITH_XPRS="_xprs" || WITH_XPRS=""
+          SIRIUS=${{ matrix.sirius }}
+          [ $SIRIUS == "ON" ] && WITH_SIRIUS="_sirius" || WITH_SIRIUS=""
+          OS="_${{ matrix.os }}"
+          APPENDIX="${OS}${WITH_SHARED}${WITH_SIRIUS}${WITH_XPRS}"
+          echo "::set-output name=appendix::$APPENDIX"
+
+      - name: Get release
+        if: github.event_name == 'release' && github.event.action == 'created'
+        id: get_release
+        uses:
+          bruceadams/get-release@v1.2.3
+
+      - name: Checkout or-tools
+        uses: actions/checkout@v2
+
+      - name: ccache
+        if: ${{ startsWith(matrix.os, 'ubuntu') }}
```

Comment:
Have you tried enabling `ccache` on Windows too ?

Reply:
yes, cmake does not like it

Reply:
We might try to use the cache for other steps, but the gain is probably not worth the effort